velox/vector/DecodedVector.cpp

```diff
@@ -105,9 +105,13 @@ void DecodedVector::makeIndices(
 
 void DecodedVector::reset(vector_size_t size) {
   if (!indicesNotCopied()) {
+    copiedIndices_.resize(size);
```

Comment:
the resize on std::vector does not change the capacity so this will not free up memory. You might have to destroy the DecodedVector if its absolutely necessary to reset the state here.